parser-typechecker/src/Unison/TermPrinter.hs

```diff
@@ -391,7 +391,8 @@ prettyPattern n c@(AmbientContext { imports = im }) p vs patt = case patt of
   l :: IsString s => String -> s
   l = fromString
   patterns vs (pat : pats) =
-    let (printed     , tail_vs      ) = prettyPattern n c (-1) vs pat
+    let (printed     , tail_vs      ) =
+          prettyPattern n c (if Pattern.application pat then 11 else -1) vs pat
```

Comment:
The substantive change is here. The rest is mostly removing unnecessary Haskell code.